scripts/install

```diff
@@ -20,166 +20,193 @@ ynh_abort_if_errors
+
 fullname=$(ynh_user_get_info "$admin" "fullname")
 mail=$(ynh_user_get_info "$admin" "mail")
 
 # Permission should be correctly set before to do this
-ynh_exec_as "$app" php $ls_cli install "$admin" "$(ynh_string_random 24)" "$fullname" "$mail"
-
-#=================================================
-# STORE THE CHECKSUM OF THE CONFIG FILE
-#=================================================
+ynh_exec_as "$app" php $ls_cli install "$admin" "$password" "$fullname" "$mail"
 
-# Calculate and store the config file checksum into the app settings
-ynh_store_file_checksum "$final_path/application/config/config.php"
+# Permission should be correctly set before to do this
+#ynh_exec_as "$app" php $ls_cli install "$admin" "$(ynh_string_random 24)" "$fullname" "$mail"
 
 #=================================================
 # LOAD SQL SPECIFIC CONFIG
 #=================================================
-ynh_configure data.sql ./data.sql
-mysql -u $db_user -p$db_pwd $db_user < ./data.sql
+ynh_add_config --template="../conf/data.sql" --destination="./data.sql"
 
-#=================================================
-# Add nice themes
-#=================================================
-#ynh_setup_source "$final_path/upload/templates/libreform" libreform
-#ynh_setup_source "$final_path/upload/templates/librepoll" librepoll
+ynh_mysql_connect_as --user="$db_user" --password="$db_pwd" --database="$db_name" < ./data.sql
 
+ynh_secure_remove --file=./data.sql
 
 #=================================================
-# GENERIC FINALISATION
-#=================================================
-# SECURING FILES AND DIRECTORIES
+# Add nice themes
 #=================================================
-ynh_script_progression --message="Apply permissions..." --time --weight=1
-# Set permissions
-set_permissions
+
+#ynh_setup_source "$datadir/upload/templates/libreform" libreform
+#ynh_setup_source "$final_path/upload/templates/librepoll" librepoll
 
 #=================================================
 # SETUP FAIL2BAN
 #=================================================
-# ynh_script_progression --message="Configuring fail2ban..." --time --weight=1
-#ynh_add_fail2ban_config "/var/log/nginx/${domain}-error.log" "PHP message: Leed: wrong login for .* client: <HOST>" 5
+#ynh_script_progression --message="Configuring fail2ban..." --weight=1
+
+# No message in logs and the apps contains already a 10 minutes ban policy
 
 #=================================================
 # SETUP SSOWAT
 #=================================================
-ynh_script_progression --message="Configuring SSOwat..." --time --weight=1
+ynh_script_progression --message="Configuring permissions..." --weight=1
+
+if [ $is_public -eq 1 ]
+then
+    allowed_groups="visitors $admin"
+else
+    allowed_groups="$admin"
+fi
+ynh_permission_update --permission="main" --show_tile="false" --add="visitors"
+
+ynh_permission_create --permission="admin" --url="/admin" --allowed=$allowed_groups  --show_tile="true"
```

Comment:
```suggestion
ynh_permission_create --permission="admin" --url="/admin" --allowed="$allowed_groups"  --show_tile="true"
```

(I doubt this works without quotes if `allowed_groups` contains spaces)

Reply:
I quote here the documentation:
> Usage: ynh_permission_create --permission="permission" [--url="url"] [--additional_urls="second-url" [ "third-url" ]] [--auth_header=true|false] [--allowed=group1 [ group2 ]] [--label="label"] [--show_tile=true|false] [--protected=true|false]

But you may be right...

Reply:
I checked and it works like i wrote it (and i don't know if it works with the double quotes)